### ISSUE TITLE
Spinner: required+min does not allow a NULL start value

### ISSUE THREAD

**Issue by melloware** [primefaces/primefaces#12352]
### Describe the bug

Reported by one of my customers.  They need to be able to specify `min="0" max="100" required="true"` and FORCE the user to pick a value.  Right now that is not possible.

here is the scenario:

```xml
<p:spinner id="allowsNull" value="#{spinnerView.number1}" required="true" max="100" />

<p:spinner id="notAllowsNull" value="#{spinnerView.number1}" required="true" min="0" max="100"/>
```

Renders like this:
![image](https://github.com/user-attachments/assets/c173d140-2f27-4595-bbbb-e083b3fc1752)

The reason is this code in the spinner.js widget has been there for years but to me this is not correct behavior.

```js
if(isNaN(parsedValue)) {
            if(PrimeFaces.trim(value) === '' && this.cfg.min !== undefined && this.cfg.required) {
                parsedValue = this.cfg.min;
            } else {
                parsedValue = null;
            }
        }
```

Basically it defaults to the `min` if `required="true"` the required becomes moot and a user can submit the page without having to select a value.

### Reproducer

[pf-spinner-required.zip](https://github.com/user-attachments/files/16428388/pf-spinner-required.zip)

### Expected behavior

The spinner should be allowed to be NULL if using a `min` and `required`

### PrimeFaces edition

Community

### PrimeFaces version

14.0.3

### Theme

ALL

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

11

**Comment by melloware** [comment]
@tandraschko @jepsar @Rapster @christophs78 would like your thoughts on this?  I guess i understand what its trying to do but I don't like the user can Submit a value without making a selection.

**Comment by jepsar** [comment]
Yeah, that's a bug in my book. If a user really wants to start with the minimum value instead of null, they can do that in the bean.

**Comment by OBUTTERLIN** [comment]
If a component is defined as "required" it means it must have a value.
Setting a min attribute has no impact on that.
The js level check should not take into account a min attribute (and why not a max if it was the only one?).

For me, the algorithm should be as follows:

If the component has no value and the required="true" attribute -> error
Otherwise if the min attribute is set and the value is less than min -> error
Otherwise if the max attribute is set and the value is greater than max -> error

The initialization ​of a bean property is to be done in the bean (via @PostConstruct for example).
Assuming that if there is no value entered but there is a min attribute and this min value should serve as the default value is nonsense.

**Comment by melloware** [comment]
@OBUTTERLIN we are on the same page and that is what my PR fixes.
